templates/sls-app-template-java/src/main/java/com/slf/util/EnvironmentConfig.java

```diff
@@ -33,18 +33,25 @@
 	public EnvironmentConfig (Context context) throws Exception {
 		String fnName = context.getFunctionName();
 		String fnNameNoEnv = null;
+		String fnNoPrefix = null;
 
 		if(null != fnName) {
 			int lastIndx = fnName.lastIndexOf("-");
 			stage = fnName.substring(lastIndx+1);
 			fnNameNoEnv = fnName.substring(0, lastIndx);
+			int preIndx = fnNameNoEnv.lastIndexOf("-");
```

Comment:
Here we are assuming that the function name itself doesn't contain a '-' character. A lambda function can contain '-': https://docs.aws.amazon.com/lambda/latest/dg/API_CreateFunction.html#SSS-CreateFunction-request-FunctionName

Here fnNoPrefix is likely to get truncated at the '-'. So a function with name: "my-function", will probably just have fnNoPrefix: "function" instead of "my-function".

Reply:
Ahhh, this a very good observation. Though honestly speaking I was continuing on the pattern of using the dash as a valid separator.